routes/goldap.js

```diff
@@ -9,17 +9,82 @@ const filer = require('../routes/file.js');
 var utils= require('./utils');
 
 /*
-var options = {
-    uri: 'ldap://'+CONFIG.ldap.host, // string
-    //version: 3, // integer, default is 3,
-    //starttls: false, // boolean, default is false
-    connecttimeout: -1, // seconds, default is -1 (infinite timeout), connect timeout
-    //timeout: 5000, // milliseconds, default is 5000 (infinite timeout is unsupported), operation timeout
-    //reconnect: true, // boolean, default is true,
-    //backoffmax: 32 // seconds, default is 32, reconnect timeout
-};
+  var options = {
+  uri: 'ldap://'+CONFIG.ldap.host, // string
+  //version: 3, // integer, default is 3,
+  //starttls: false, // boolean, default is false
+  connecttimeout: -1, // seconds, default is -1 (infinite timeout), connect timeout
+  //timeout: 5000, // milliseconds, default is 5000 (infinite timeout is unsupported), operation timeout
+  //reconnect: true, // boolean, default is true,
+  //backoffmax: 32 // seconds, default is 32, reconnect timeout
+  };
 */
 
+
+function get_group_dn(group) {
+    return new Promise( function (resolve, reject) {
+        // todo: factorize with get_user_dn
+        let client = myldap.createClient({
+            url: 'ldap://' +  CONFIG.ldap.host
+        });
+        client.bind(CONFIG.ldap.admin_cn + ',' + CONFIG.ldap.admin_dn, CONFIG.ldap.admin_password, function(err) {
+            if(err) {
+                logger.error('Failed to bind as admin to ldap', err);
+                reject(err);
+                return;
+            }
+        });
+
+        let group_dn_list = [];
+
+        let opts = {
+            filter: '(cn=' + group + ')',
+            scope: 'sub',
+            attributes: ['dn']
+        };
+
+        client.search('ou=groups,' + CONFIG.ldap.dn, opts, function(err, res) {
+            if(err) {
+                logger.error('Could not find ' + group, err);
+                reject(err);
+                return;
+            }
+
+            res.on('searchEntry', function(entry) {
+                logger.info('dn found: ', entry.object['dn'].trim(), ' for ', group);
+                group_dn_list.push(entry.object['dn'].trim());
+            });
+            res.on('error', function(err) {
+                logger.error('error ', err);
+                reject(err);
+                return;
+            });
+            // eslint-disable-next-line no-unused-vars
+            res.on('end', function(result) {
+                let default_dn = 'cn=' + group + ',ou=groups,' + CONFIG.ldap.dn;
+
+                if (group_dn_list.length > 1) {
+                    logger.error('more than one entry have been found', group_dn_list);
```

Comment:
could you record an event with the error when this case if found , in addition to log.

Reply:
I think event are related (and attached) to user as owner, here we don't know the user, do you want an event with admin as owner ?

Reply:
194d0e707f7e4717af1db35f65d2cf8472efa692